srcs/juloo.keyboard2/EmojiGridView.java

```diff
@@ -110,6 +111,33 @@ SharedPreferences emojiSharedPreferences()
     return getContext().getSharedPreferences("emoji_last_use", Context.MODE_PRIVATE);
   }
 
+  private void migrateOldPrefs()
+  {
+    final String MIGRATION_CHECK_KEY = "MIGRATION_COMPLETE";
+
+    SharedPreferences prefs;
+    try { prefs = emojiSharedPreferences(); }
+    catch (Exception e) { return; }
+
+    Set<String> lastUsed = prefs.getStringSet(LAST_USE_PREF, null);
+    if (lastUsed != null && !prefs.getBoolean(MIGRATION_CHECK_KEY, false))
+    {
+      SharedPreferences.Editor edit = prefs.edit();
+      edit.clear();
+
+      Set<String> lastUsedNew = new HashSet<>();
+      for (String entry : lastUsed)
+      {
+        String[] data = entry.split("-", 2);
+        lastUsedNew.add(Integer.parseInt(data[0]) + "-" + Emoji.mapOldNameToValue(data[1]));
```

Comment:
`mapOldNameToValue` can throw an exception and it's not fine to crash the app for such a small feature.
I think catching the exception here and not adding the problematic emoji to `lastUsedNew` is a nice way to handle it.

Reply:
Whoops, thanks for noticing that. I made it throw for that exact purpose and forgot to actually catch it